itachallenge-user/src/test/java/com/itachallenge/user/controller/UserControllerTest.java

```
@@ -177,6 +231,24 @@ void addToFavorites_WhenBadFormattedId_Returns404() {
         verify(userService, times(1)).addChallengeToFavorites(userId, challengeId);
     }
 
+    @Test
+    void addToBookmarks_WhenBadFormattedId_Returns404() {
+        String userId = "invalidUuid";
+        String challengeId = "invalidUUid";
+        when(userService.addChallengeToBookmarks(userId, challengeId))
+                .thenReturn(Mono.error(new BadUUIDException("Error message")));
+
+        webTestClient.post()
+                .uri("/itachallenge/api/v1/user/users/" + userId + "/bookmarks/" + challengeId)
+                .exchange()
+                .expectStatus().isEqualTo(HttpStatus.BAD_REQUEST)
+                .expectHeader().valueEquals("X-Bookmark-Added", "False")
+                .expectHeader().valueEquals("X-Bookmark-Message", "The provided IDs are not valid.")
+                .expectBody(Boolean.class).isEqualTo(false);
+
+        verify(userService, times(1)).addChallengeToBookmarks(userId, challengeId);
+    }
+
```

the endpoint should return a 400 error, right? Now checking it I saw that Juanra also had it like that

---

Well Spotted! Changed!